src/main/webapp/app/exercises/programming/shared/code-editor/build-output/code-editor-build-output.component.ts

```diff
@@ -182,15 +183,14 @@ export class CodeEditorBuildOutputComponent implements AfterViewInit, OnInit, On
 
     /**
      * Decides if the build log should be shown.
-     * If All tests were successful or there is test feedback -> don't show build logs.
-     * Else -> show build logs.
+     * Fetch the build logs only if the submission could not be build
      * @param result
      */
     fetchBuildResults(result: Result | null): Observable<BuildLogEntry[] | null> {
-        if ((result && result.successful) || (result && !result.successful && result.feedbacks && result.feedbacks.length)) {
```

Comment:
is this logic not necessary any more?

Reply:
The whole feedback available logic is what caused the problems in the first place. I also think `submission.isBuildFailed` is a stronger check than `result.successful`. Added a check for the result. If no result is available we don't need to check for build logs.